plugins/inputs/mongodb/mongodb.go

```diff
@@ -164,6 +168,24 @@ func (m *MongoDB) Gather(acc telegraf.Accumulator) error {
 		wg.Add(1)
 		go func(srv *Server) {
 			defer wg.Done()
+			if !srv.reachable {
+				// is not reachable try a reconnect
+				disconnectCtx, disconnectCancel := context.WithTimeout(context.Background(), 1*time.Second)
+				defer disconnectCancel()
+				err := srv.client.Disconnect(disconnectCtx)
+				if err != nil {
+					m.Log.Errorf("unable to reconnect to MongoDB: %q", err)
+					return
+				}
+				connectCtx, connectCancel := context.WithTimeout(context.Background(), 1*time.Second)
+				defer connectCancel()
```

Comment:
Reusing the connect context from init might be a good idea, is there a reason you picked `1*time.Second` for the timeout?

Reply:
No reason for 1 second, it was just looking like a simple default.